pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1554 @@
+	}
+
+	if hasScopes, success := g.(interface {
+		GetScopes() *openapi_v2.Oauth2Scopes
+	}); success {
+		scopes := hasScopes.GetScopes()
+		if scopes != nil {
+			k.Scopes = make(map[string]string, len(scopes.AdditionalProperties))
+			for _, v := range scopes.AdditionalProperties {
+				k.Scopes[v.Name] = v.Value
+			}
+		}
+	}
+
+	return ok, nil
+}
+
+func (k *SecurityScheme) FromGnostic(g *openapi_v2.SecurityDefinitionsItem) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+	ok = true
+	switch s := g.Oneof.(type) {
+	case *openapi_v2.SecurityDefinitionsItem_ApiKeySecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.ApiKeySecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.ApiKeySecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_BasicAuthenticationSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.BasicAuthenticationSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.BasicAuthenticationSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_Oauth2AccessCodeSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.Oauth2AccessCodeSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.Oauth2AccessCodeSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_Oauth2ApplicationSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.Oauth2ApplicationSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.Oauth2ApplicationSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_Oauth2ImplicitSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.Oauth2ImplicitSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.Oauth2ImplicitSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_Oauth2PasswordSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.Oauth2PasswordSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.Oauth2PasswordSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	}
```

Comment:
needs a default case that returns an error, right?

Reply:
yup